pkg/resource/stack/deployment.go

```diff
@@ -373,12 +373,16 @@ func SerializePropertyValue(prop resource.PropertyValue, enc config.Encrypter,
 	// We serialize resource references using a map-based representation similar to assets, archives, and secrets.
 	if prop.IsResourceReference() {
 		ref := prop.ResourceReferenceValue()
-		return map[string]interface{}{
+
+		serialized := map[string]interface{}{
 			resource.SigKey:  resource.ResourceReferenceSig,
 			"urn":            ref.URN,
-			"id":             ref.ID,
 			"packageVersion": ref.PackageVersion,
```

Comment:
Nit: I noticed we don't include `packageVersion` in the gRPC marshaling if it's an empty string. Should we do the same here?

https://github.com/pulumi/pulumi/blob/036c4205fd7e84abab5711e2d223ea4729520037/sdk/go/common/resource/plugin/rpc.go#L175-L184

Reply:
I'd rather leave that as-is for now to avoid touching more code than necessary.